noir-projects/aztec-nr/aztec/src/context/public_context.nr

```diff
@@ -58,7 +58,7 @@ impl PublicContext {
     pub fn new(inputs: PublicContextInputs, args_hash: Field) -> PublicContext {
         PublicContext {
             inputs,
-            side_effect_counter: inputs.start_side_effect_counter,
+            side_effect_counter: inputs.start_side_effect_counter + 1,
```

Comment:
That's a good point! Modified [here](https://github.com/AztecProtocol/aztec-packages/pull/6365/files#diff-e5cdf0f9189808523b4355c7cde49a909eb6544ab578f9a73b60b5fb0dae9878R86) to be consistent with the private land. 
We are not checking the counters in public kernel. And it won't necessary have to +1 for AVM. It's incremented by 1 because it's easier for private kernel to validate if the counters are strictly increasing.